docs/general/dql/union.rst

```diff
@@ -77,4 +82,25 @@ To remove duplicates, use ``UNION DISTINCT`` or simply ``UNION``.
     SELECT 7 rows in set (... sec)
 
 
-.. _Union (SQL): https://en.wikipedia.org/wiki/Set_operations_(SQL)#UNION_operator
\ No newline at end of file
+.. _Union (SQL): https://en.wikipedia.org/wiki/Set_operations_(SQL)#UNION_operator
+
+.. _union-of-object-types:
+
+Union of object types
+---------------------
+
+When object types are unioned, the resulting object will contain sub-columns
+from both of the input objects.
+
+::
+
+    cr> SET error_on_unknown_object_key = FALSE;
+    SET OK, 0 rows affected  (... sec)
+    cr> SELECT o, o['a'], o['b'] FROM (SELECT {a=1} AS o UNION SELECT {b=1} AS o) AS t;
```

Comment:
maybe add a `c` subcol on both sides one string one numeric to show the merging and cast?

Reply:
Actually, object literals' inner data types are `UndefinedTypes` currently(should be a bug but probably not high priority to fix). I have added `Union of different types` instead.